mlir/lib/Quantum/Utils/RemoveQuantum.cpp

```diff
@@ -59,6 +59,23 @@ void removeQuantumMeasurements(func::FuncOp &function, PatternRewriter &rewriter
     }
 }
 
+void replaceQuantumMeasurements(func::FuncOp &function, PatternRewriter &rewriter)
+{
+    std::deque<Operation *> opsToReplace;
+    function.walk([&](MeasurementProcess op) { opsToReplace.push_back(op); });
+    for (auto op : opsToReplace) {
```

Comment:
I don't think we need the deque, we can just do the replacement of the ops in the walk no, or does that create issues? 🤔

Reply:
Good catch it is from the other function, I will remove it.

Reply:
Just wanted to double check, indeed it is okay to erase the matched op as long as we are doing post-order (the default):
>  /// Walk all of the regions, blocks, or operations nested under (and including)
 /// the given operation. The order in which regions, blocks and operations at
 /// the same nesting level are visited (e.g., lexicographical or reverse
 /// lexicographical order) is determined by 'Iterator'. The walk order for
 /// enclosing regions, blocks and operations with respect to their nested ones
 /// is specified by 'order'. These methods are invoked for void-returning
 /// callbacks. **A callback on a block or operation is allowed to erase that block
 /// or operation only if the walk is in post-order. See non-void method for
 /// pre-order erasure.**